### ISSUE TITLE
vsce doesn't respect engines versions in package.json

### ISSUE THREAD

**Issue by ws93** [Microsoft/vscode-vsce#339]
Hi team,  I was trying to generate `vsix` file for my extension. In extension source code, I did a version check on a dependency(or `engine`, also a npm package) using [`semver`.](https://www.npmjs.com/package/semver). Recently I changed this version check to make it pass the pre-release of that dependency. It worked fine when I tried in debug mode(Extension Host). But the version check failed when I was using generated vsix file. The error message from extension still tells me an old version of that dependency I specified before.
So I just wonder if `vsce` doesn't get `engines` version from `package.json`, then where does this information come from?

**Comment by joaomoreno** [comment]
I'm not entirely sure I understand the problem.

Can you give me specific steps to reproduce it?

**Comment by ws93** [comment]
I will try. This is an internal tool in our company so you can't find the specific npm package but I'll try to give you the steps that I had an issue.

0. Let's call our extension `myExt` and call that dependency(engine) `depen`
1. in `package.json`'s `engines` field, specify `depen` like this
```
    "engines": {
        "vscode": "^1.27.0",
        "depen": ">=0.5.0"
    },
```
2. in `myExt` source code, write some code to get `depen` version from `package.json`'s `engines` field.
3. change `depen` version in `package.json`. For example `>=0.6.0`
4. debug `myExt`, found that line in Step 2 gives me new version `>=0.6.0`
5. use `vsce` to generate `vsix` file and install this `vsix`. Use this extension and found it still gives me old version `>=0.5.0`

**Comment by ws93** [comment]
Hope it helps to understand.

**Comment by joaomoreno** [comment]
A VSIX is a ZIP file. Rename its extension to ZIP and extract it. In there you'll find `package.json`. Does it have the old or the new version?

**Comment by ws93** [comment]
Thanks for your help. The version in vsix file is new.
I found one thing interesting. When I install an extension, vscode will put this extension under `%temp%/.vscode/extensions/`. And even I uninstall this extension, vscode will still keep those fiiles. So then when I install a new version of this extension, there's some discrepancy. Because it looks like vscode is not using what I just installed but the cached extension in temp folder.
Looks like vscode has some strange behavior.
